### DISCUSSION TITLE
Should external deployments connect do DB via internal credential?

### DISCUSSION THREAD

**Discussion by johngrimsey** [Dokploy/dokploy#529] [Q&A] [answered]
I'm not sure if docker/docker-compose services deployed to external service should use internal or external DB credentials for Postgres database application hosted on external server.

My understanding is that as Dokploy uses swarm under the hood, all containers are run on the same network and should be able to reference each other using internal names - regardless of whether they're on external server or Dokploy server.

Am I right?

**Comment by johngrimsey** [comment]
@Siumauricio Sorry for all the questions! Still setting everything up here.

**Comment by Siumauricio** [comment, marked as answer]
the remote servers are isolated instances of the main instance of dokploy, so you could not connect internally, you would have to expose your database in case you have services on other servers and want to connect, but if the services are on the same server you can connect internally.

**Reply by johngrimsey** [reply]
Thanks!
